src/lib/FreeC/Pass/ExportPass.hs

```diff
@@ -103,3 +113,23 @@ exportInterface modName = do
   isExported entry = case entryName entry of
     IR.Qual modName' _ -> modName' == modName
     IR.UnQual _        -> False
+
+  -- Qualifies the Coq identifier in the module interface with the module name
+  -- when an entry of the module is exported.
+  -- This ensures that any module that imports the entry uses its qualified
+  -- name, which prevents name conflicts between imported modules.
```

Comment:
Use Haddock comment also for local functions ([styleguide](https://github.com/FreeProving/guidelines/blob/main/CONTRIBUTING.md#haddock-styleguide)).